geonode/maps/tests.py

```diff
@@ -779,6 +824,82 @@ def test_batch_edit(self):
                 self.assertTrue(word.name in keywords.split(','))
 
 
+class InitialSetup():
+
+    @classmethod
+    def generate_initial_map(cls):
+        # construct json request for new map
+        json_payload = {
+            "sources": {
+                "source_OpenMapSurfer Roads": {
+                    "url": "http://korona.geog.uni-heidelberg.de/tiles"
+                           "/roads/x={x}&y={y}&z={z}"
+                },
+                "source_OpenStreetMap": {
+                    "url": "http://{s}.tile.osm.org/{z}/{x}/{y}.png"
+                },
+                "source_san_andres_y_providencia_administrative": {
+                    "url": "http://geonode.dev/qgis-server/tiles"
```

Comment:
Can you take this kind of URL dynamically from the layer object instead of hardcoded it.